qiskit_experiments/data_processing/nodes.py

```diff
@@ -56,130 +106,266 @@ def _format_data(self, datum: Any) -> Any:
+            The mean that was determined during training for the given qubit and IQ index.
         """
-        if self.scale is None:
-            return datum[:, :, 0]
+        return self._means[qubit][iq_index]
 
-        return datum[:, :, 0] * self.scale
+    @property
+    def scales(self) -> List[float]:
+        """Return the scaling of the SVD."""
+        if self._scales:
+            return self._scales
 
+        raise DataProcessorError("SVD is not trained.")
 
-class ToRealAvg(IQPart):
-    """IQ data post-processing. Isolate the real part of averaged IQ data."""
+    @property
+    def is_trained(self) -> bool:
+        """Return True is the SVD has been trained.
 
-    def _required_dimension(self) -> int:
-        """Require memory to be a 2D array."""
-        return 2
+        Returns:
+            True if the SVD has been trained.
+        """
+        return self._main_axes is not None
 
-    def _process(self, datum: np.array) -> np.array:
-        """Take the real part of the IQ data.
+    def _process(
+        self, datum: np.array, error: Optional[np.array] = None, **options
+    ) -> Tuple[np.array, np.array]:
+        """Project the IQ data onto the axis defined by an SVD and scale it.
 
         Args:
-            datum: A 2D array of qubits, and a complex averaged IQ point as [real, imaginary].
+            datum: A 2D array of qubits, and an average complex IQ point as [real, imaginary].
+            error: An optional 2D array of qubits, and an error on an average complex IQ
+                point as [real, imaginary].
 
         Returns:
-            A 1D array. Each entry is the real part of the averaged IQ data of a qubit.
+            A Tuple of 1D arrays of the result of the SVD and the associated error. Each entry
+            is the real part of the averaged IQ data of a qubit.
+
+        Raises:
+            DataProcessorError: If the SVD has not been previously trained on data.
         """
-        if self.scale is None:
-            return datum[:, 0]
 
-        return datum[:, 0] * self.scale
+        if not self.is_trained:
+            raise DataProcessorError("SVD must be trained on data before it can be used.")
 
+        n_qubits = datum.shape[0]
+        processed_data = []
 
-class ToImag(IQPart):
-    """IQ data post-processing. Isolate the imaginary part of single-shot IQ data."""
+        if error is not None:
+            processed_error = []
+        else:
+            processed_error = None
 
-    def _required_dimension(self) -> int:
-        """Require memory to be a 3D array."""
-        return 3
+        # process each averaged IQ point with its own axis.
+        for idx in range(n_qubits):
+
+            centered = np.array(
+                [datum[idx][iq] - self.means(qubit=idx, iq_index=iq) for iq in [0, 1]]
+            )
+
+            processed_data.append((self._main_axes[idx] @ centered) / self._scales[idx])
```

Comment:
Why we don't use property here (seems like this is only place it can be used)? If the property is not necessary probably we can remove.
```suggestion
            processed_data.append((self._main_axes[idx] @ centered) / self.scales[idx])
```

Reply:
The rational for having the scales in as a property is to let users see how the SVD was trained. Note that I also switched to the property as you suggest. See here: 1a79dcd